packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2774,7 +2783,7 @@ class Navigator extends StatefulWidget {
   /// [Route.didChangeNext] and [Route.didChangePrevious]). If the [Navigator]
   /// has any [Navigator.observers], they will be notified as well (see
   /// [NavigatorObserver.didRemove]). The removed route is disposed without
-  /// being notified. The future that had been returned from pushing that routes
+  /// being notified. The future that had been returned from pushing that route
   /// will not complete.
```

Comment:
It does feel a bit weird that `removeRouteBelow` is no longer consistent with `removeRoute`. Should we change all of these methods to complete their respective futures?

Reply:
pushAndRemoveUntil, restorablePushAndRemoveUntil, restorablePushNamedAndRemoveUntil and pushAndRemoveUntil could be supported but we need to add an extra logique like a callback to let developper choose to override route the value of `didComplete(result)`. if the callback is not provided we can fallback to null.

Reply:
PR edited to support removeRouteBelow.

Reply:
We can also go forward and try to support all method like replace, restorable etc.

Reply:
Must i support them all inside this PR with the risk to delay massively it or should i create a dedicated PR later ? (New unit test will be needed, lot of changes)

Reply:
> Must I support them all inside this PR with the risk of a massive delay, or should I create a dedicated PR later?

That's a really good question.

I suppose my question would be: is it all right for `removeRoute` & `removeRouteBelow` to both complete their respective futures, but for none of the other similar methods to do this?

If we merge this pull request today and there end up being issues with migrating the similar methods, I think the API would be more inconsistent than it is now.

Reply:
> pushAndRemoveUntil, restorablePushAndRemoveUntil, restorablePushNamedAndRemoveUntil and pushAndRemoveUntil could be supported but we would need to add extra logic, like a callback, to give the developer an option to override the value of `didComplete(result)`; if the callback is not provided we can fallback to null.

My opinion as of now is that it'd be best for each removed route to have its future completed, but we can just do `didComplete(null)` for now. If we want `pushAndRemoveUntil` to support completing with a value, that can be saved for a future PR.

Does that sound okay to you?

Reply:
That could be a great roadmap : first time didComplete(null) by default, next PR logic to well handle that everywhere.

Reply:
I tested quickly on my side, I can well update all these functions to support the `result` property, and when needed, a dedicated callback for whose related to popUntil and removeUntil. Will do that later in dedicated PR (if you and reviewer agree).

Reply:
SGTM!